### PR TITLE
Enhancement/improve merchant payment page

### PR DESCRIPTION
**Pull request by Aierie**
This PR addresses:
- [CS-1830]: makes the url generated on the merchant payment request page start with `https://wallet.cardstack.com` in production, and `https://wallet-staging.stack.cards` in staging and dev. Did not add these as constants to the SDK for now due to difficulties getting imports to work.
- [CS-1837]: makes incomplete routes (eg. `https://wallet.cardstack.com/pay/sok`) match the `pay-error` page.

### CODE REVIEW

File: packages/web-client/app/router.ts [hunk 1]
```diff
@@ -16,5 +16,8 @@ Router.map(function () {
   this.route('pay', {
     path: '/pay/:network/:merchant_safe_id',
   });
+  this.route('pay-error', {
```
Comment (jurgenwerk) [inline packages/web-client/app/router.ts:19]: Would it be better to name this `pay-missing-route` or something like that? Just to make it a bit clearer.

**Review by jurgenwerk — APPROVED** [review]
LGTM! Just a very minor comment.

File: packages/web-client/app/router.ts (hunk 1 above)
Comment (Aierie) [reply packages/web-client/app/router.ts:19]: At the moment this is reusing Ember's error substate route (that is rendered when we fail to fetch a merchant safe) https://guides.emberjs.com/release/routing/loading-and-error-substates/#toc_error-substates, is there a good way to specify that without creating a new route? Should we create a new route for this case?

Comment (lukemelia) [reply packages/web-client/app/router.ts:19]: Having a declared route named the same as implicit routes like this should be avoided, IMO

Comment (lukemelia) [reply packages/web-client/app/router.ts:19]: You can use the `templateName` property of a route to reuse the template.

File: packages/web-client/app/routes/pay-missing-route.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import Route from '@ember/routing/route';
+import '../css/pay.css';
+
+export default class PayErrorRoute extends Route {
```
Comment (lukemelia) [inline packages/web-client/app/routes/pay-missing-route.ts:4]: Rename this class name for clarity.

Comment (Aierie) [reply packages/web-client/app/routes/pay-missing-route.ts:4]: Oops, thanks!

**Pull request merged**